src/main/java/org/openstreetmap/atlas/generator/tools/json/PersistenceJsonParser.java

```diff
@@ -0,0 +1,169 @@
+package org.openstreetmap.atlas.generator.tools.json;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.sharding.Shard;
+import org.openstreetmap.atlas.utilities.collections.Maps;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * Define a JSON format for the RDD keys used by the spark job. This class provides methods for
+ * constructing the key as well as parsing data out of the key. The JSON format should follow this
+ * structure:
+ * 
+ * <pre>
+ * {
+ *    "country": "USA-0-generated",
+ *    "shard": "1-2-3",
+ *    "metadata": {
+ *         "scheme": "zz/",
+ *    }
+ * }
+ * </pre>
+ * 
+ * @author lcram
+ */
+public final class PersistenceJsonParser
```

Comment:
Seems this parse is trying to read schema file and covert it to json file? if that is the case, check out ObjectMapper, it maybe the thing you need.

Reply:
This class exists just to enforce the JSON structure required by `AbstractMultipleAtlasBasedOutputFormat`. So you give it a country, shard, and scheme, and it gives you a JSON key that the output format class understands. I tagged you in a spot where we are using it in atlas-generator. Downstream code should emulate this if they are using saveAsHadoopFile with atlases.